web-app/react-app/tests/utils/web3Provider.test.js

```diff
@@ -53,8 +53,26 @@ describe('web3Provider', () => {
         })
     })
     describe('getWeb3', () => {
-        it('must return the same instance every time', () => {
-            assert(getWeb3() === getWeb3())
+        beforeEach(() => {
+            global.web3 = undefined
+            global.ethereum = undefined
+        })
+        it('must return the web3 object without a provider when metamask does not provide it', () => {
+            const web3 = getWeb3()
+            expect(web3.currentProvider).toEqual(null)
+        })
+        it('must return the web3 object with the window.web3.currentProvider provider if it is available/defined', () => {
+            // 'legacy' metamask web3 injection scenario
+            global.web3 = Web3
```

Comment:
Technically `global.web3` shouldn't be `Web3`, but `new Web3()`. Not a big thing, though. (and also using just an object would be enough)